scout/adapter/mongo/case.py

```diff
@@ -108,13 +102,21 @@ def _set_similar_phenotype_query(
 
             similar_cases = self.cases_by_phenotype(hpo_terms, institute_id, case_obj["_id"])
         else:  # similar HPO terms
-            hpo_terms = list(query_term.replace(" ", "").split(","))
+            hpo_terms = list(query_value.replace(" ", "").split(","))
             similar_cases = self.cases_by_phenotype(hpo_terms, institute_id, None)
 
-        query["_id"] = {"$in": [similar_case[0] for similar_case in similar_cases]}
+        similar_case_ids = [
+            similar_case[0] for similar_case in similar_cases
+        ]  # These are sorted by the most similar to the least
+        self._update_case_id_query(
```

Comment:
This advanced case search is quite delicate, because if you have other params that will affect the content of query["_id"]["$in"] you can mess up the order of the similar cases